server/src/test/java/com/google/coffeehouse/servlets/CreateClubServletTest.java

```diff
@@ -0,0 +1,174 @@
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+
+/**
+ * Unit tests for {@link CreateClubServlet}.
+ */
+public class CreateClubServletTest {
+  private static final String NAME = "Club Name";
+  private static final String DESCRIPTION = "Club Description";
+  private static final String IDENTIFICATION_STRING = "predetermined-identification-string";
+  private static final String BOOK_TITLE = "Book Name";
+  private static final String BOOK_AUTHOR = "Book Author";
+  private static final String BOOK_ISBN = "978-3-16-148410-0";
+  private static final String MINIMUM_JSON = String.join("\n", 
+      "{",
+      "  \"name\" : \"" + NAME + "\",",
+      "  \"currentBook\" : {",
+      "    \"title\" : \"" + BOOK_TITLE + "\"",
+      "  }",
+      "}");
+  private static final String MAXIMUM_JSON = String.join("\n", 
+      "{",
+      "  \"name\" : \"" + NAME + "\",",
+      "  \"description\" : \"" + DESCRIPTION + "\",",
+      "  \"contentWarnings\" : [\"1\", \"2\"],",
+      "  \"currentBook\" : {",
+      "    \"title\" : \"" + BOOK_TITLE + "\",",
+      "    \"author\" : \"" + BOOK_AUTHOR + "\",",
+      "    \"isbn\" : \"" + BOOK_ISBN + "\"",
+      "  }",
+      "}");
+  private static final String NO_NAME_JSON = String.join("\n", 
+      "{",
+      "  \"currentBook\" : {",
+      "    \"title\" : \"" + BOOK_TITLE + "\"",
+      "  }",
+      "}");
+  private static final String NO_BOOK_JSON = String.join("\n", 
+      "{",
+      "  \"name\" : \"" + NAME + "\"",
+      "}");
+  private static final String SYNTACTICALLY_INCORRECT_JSON = "{\"name\"";
+
+  private CreateClubServlet CreateClubServlet;
+  private final LocalServiceTestHelper helper = new LocalServiceTestHelper();
+  private StringWriter stringWriter = new StringWriter();
+
+  @Mock private HttpServletRequest request;
+  @Mock private HttpServletResponse response;
+
+  @Before
+  public void beforeTest() throws IOException {
+    helper.setUp();
+
+    IdentifierGenerator idGen = Mockito.mock(IdentifierGenerator.class);
+    Mockito.when(idGen.generateId()).thenReturn(IDENTIFICATION_STRING);
+    CreateClubServlet = new CreateClubServlet(idGen);
+
+    request = Mockito.mock(HttpServletRequest.class);
+    response = Mockito.mock(HttpServletResponse.class);
+    Mockito.when(response.getWriter()).thenReturn(new PrintWriter(stringWriter));
+  }
+
+  @After
+  public void afterTest() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void minimumValidInput() throws IOException {
+    Mockito.when(request.getReader()).thenReturn(
+          new BufferedReader(new StringReader(MINIMUM_JSON)));
+
+    CreateClubServlet.doPost(request, response);
+    String result = stringWriter.toString();
+
+    // Use multiple contains because order of JSON should not be relied upon per GSON documentation
```

Comment:
Yeah this is a good catch. I would actually recommend using GSON here to parse the resulting JSON and turn it into a map that you can test. Then it will also throw an error if the resulting JSON is invalid

Reply:
Very smart! I've replaced the `result.contains` with this approach.